Tests/Unit/MSFT_xWebAppPool.Tests.ps1

```diff
@@ -0,0 +1,862 @@
+
+if(-not (Test-Path -Path $moduleRoot))
+{
+    $null = New-Item -Path $moduleRoot -ItemType Directory
+}
+else
+{
+    # Copy the existing folder out to the temp directory to hold until the end of the run
+    # Delete the folder to remove the old files.
+    $tempLocation = Join-Path -Path $env:Temp -ChildPath $DSCModuleName
+    Copy-Item -Path $moduleRoot -Destination $tempLocation -Recurse -Force
+    Remove-Item -Path $moduleRoot -Recurse -Force
+    $null = New-Item -Path $moduleRoot -ItemType Directory
+}
+
+Copy-Item -Path $PSScriptRoot\..\..\* -Destination $moduleRoot -Recurse -Force -Exclude '.git'
+
+if (Get-Module -Name $DSCModuleName -All)
+{
+    Get-Module -Name $DSCModuleName -All | Remove-Module
+}
+
+Import-Module -Name $(Get-Item -Path (Join-Path $moduleRoot -ChildPath "$DSCModuleName.psd1")) -Force
+
+if (($env:PSModulePath).Split(';') -ccontains $pwd.Path)
+{
+    $script:tempPath = $env:PSModulePath
+    $env:PSModulePath = ($env:PSModulePath -split ';' | Where-Object {$_ -ne $pwd.path}) -join ';'
+}
+
+
+$executionPolicy = Get-ExecutionPolicy
+if ($executionPolicy -ne 'Unrestricted')
+{
+    Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Force
+    $rollbackExecution = $true
+}
+
+$ModuleName = "MSFT_xWebAppPool"
+
+Import-Module (Join-Path $moduleRoot -ChildPath "DSCResources\$ModuleName\$ModuleName.psm1")
+
+Describe "MSFT_xWebAppPool"{
+    InModuleScope $ModuleName {
+        Function TestTargetResourceSame
+        {
+            <#
+
+            .SYNOPSIS
+            Runs Test-TargetResource on $option so that when $option has the same value as the current setting, $true is returned.
+
+            .DESCRIPTION
+            Runs Test-TargetResource on $option so that when $option has the same value as the current setting, $true is returned.
+
+            .PARAMETER option 
+            should be the name of the option as defined in xWebAppPool.psm1
+
+            .PARAMETER appcmdVal
+            The location that would be used to get this value via appcmd.
+
+            .EXAMPLE
+            TestTargetResourceSame "autoStart" "add.autoStart"
+
+            #>
+            [OutputType([bool])]
+            Param($option,$appcmdVal)
+            $testParams =@{
+                Name = "DefaultAppPool"
+                Ensure = "Present"
+            }
+
+            $arrVal = $appcmdVal.Split(".")             
+            $val = $global:PoolCfg.GetElementsByTagName($arrVal[$arrVal.length-2]) | Select -ExpandProperty $arrVal[$arrVal.length-1]
+            $testParams.Add($option, $val)
+            Mock Invoke-AppCmd {return $global:PoolCfg.OuterXml}
```

Comment:
Both Invoke-AppCmd and Invoke-AppCmdState can be moved outside of this function.